PSWebServiceLibrary.php

```diff
@@ -506,15 +551,349 @@ public function delete($options)
+                    '%code%' => $code,
+                    '%status%' => $status,
+                )
+            ),
+            $code,
+            $previous
+        );
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive100ContinueStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(100, 'Continue', $previous);
+    }
+    
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive101SwitchingProtocolsStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(101, 'Switching Protocols', $previous);
+    }
+    
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive102ProcessingStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(102, 'Processing', $previous);
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive103EarlyHintsStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(103, 'Early Hints', $previous);
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive300MultipleChoicesStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(300, 'Multiple Choices', $previous);
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive301MovedPermanentlyStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(301, 'Moved Permanently', $previous);
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive302FoundStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(302, 'Found', $previous);
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive303SeeOtherStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(303, 'See Other', $previous);
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive304NotModifiedStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(304, 'Not Modified', $previous);
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive307TemporaryRedirectStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(307, 'Temporary Redirect', $previous);
+    }
+
+    /**
+     * @param ?\Throwable $previous
+     * @return self
+     */
+    public static function shouldNotReceive308PermanentRedirectStatus($previous = null)
+    {
+        return self::shouldNotReceiveStatus(308, 'Permanent Redirect', $previous);
+    }
```

Comment:
```suggestion
    /**
     * @param int $code
     * @param string $status
     * @param ?\Throwable $previous
     * @return self
     */
    private static function shouldNotReceiveStatus($code, $status, $previous = null)
    {
        static $messages = array(
            100 => 'Continue',
            101 => 'Switching protocols',
            102 => 'Processing',
            103 => 'Early Hints',
            300 => 'Multiple Choices',
            301 => 'Moved Permanently',
            302 => 'Found',
            303 => 'See Other',
            304 => 'Not Modified',
            307 => 'Temporary Redirect',
            308 => 'Permanent Redirect'
        );
        return new self(
            strtr(
                'This call to PrestaShop Web Services responded with status `%code% %status%`, this client was not designed to process this kind of response. This can come from your web server or reverse proxy configuration.',
                array(
                    '%code%' => $code,
                    '%status%' => $status,
                )
            ),
            $code,
            $previous
        );
    }
```

Reply:
That's a nice simplification of the code, however `$messages` is not used in the message output yet And it should probably be a `const` on the class instead of a static variable You can then get the message via `self::STATUS_MESSAGES[$code]` thus only one parameter would be needed for this method